lib/rubocop/cop/correctors/multiline_literal_brace_corrector.rb

```diff
@@ -42,8 +42,30 @@ def correct_next_line_brace(corrector)
 
         corrector.insert_before(
           last_element_range_with_trailing_comma(node).end,
+          content_if_comment_present(corrector, node)
+        )
+      end
+
+      def content_if_comment_present(corrector, node)
+        range = range_with_surrounding_space(
+          range: children(node).last.source_range,
+          side: :right
+        ).end.resize(1)
+        if range.source == '#'
+          select_content_to_be_inserted_after_last_element(corrector, node)
+        else
           node.loc.end.source
+        end
+      end
+
+      def select_content_to_be_inserted_after_last_element(corrector, node)
+        range = range_between(
+          node.loc.end.begin_pos,
+          range_by_whole_lines(node.loc.expression).end.end_pos
         )
+
+        corrector.remove(range)
```

Comment:
If I remove this line, all tests still pass. I suspect that this is because the call to `remove` is not needed, rather that being related to poor test coverage. But I don't see why it's not needed. :confused: 

Anyway, it's better if the method only selects a certain source range, like its name says, and not does anything else (single responsibility principle).

Reply:
@jonas054 , Thanks for reviewing, Yes, The mentioned line is needed. I wrote the wrong test case which I have corrected now. Please review now.